Implemented a method to change the data of packet.

renatocarusos opened the pull request:
This method work with char[], string, int or float variables. In float, there have an option to determine how many decimal will be, instead as three, as default.
There is an example sketch too.

renatocarusos committed:
Implemented a method to change package data.

This method work with char[], string, int or float variables. In float, there have an option to determine how many decimal will be, instead as three, as default.
There is an example sketch too.

jgromes commented on src/Packet.cpp:
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
+void Packet::setPacketData(String wordy){
+  char charArray[240];
+  wordy.toCharArray(charArray, sizeof(wordy));
+  this->setPacketData(charArray);
```
Unneccessary `this` pointer (it is implied when calling member method `setPacketData` from inside its own class).
The same goes for all `this` pointers in this file.

jgromes commented on src/Packet.cpp:
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
+  }
+}
+
+void Packet::setPacketData(float f, uint8_t decimals) {
```
This is not a good way to overload functions - you have basically the same code twice here in both implementations of `setPacketData`.

Delete the previous `setPacketData` (line 164) and in this one, add default value to the `decimals` argument in Packet.h

jgromes commented on src/Packet.h:
```diff
@@ -35,6 +35,11 @@ class Packet {
+    void setPacketData(char charArray[240]);
+    void setPacketData(String wordy);
+    void setPacketData(float f);
+    void setPacketData(float f, uint8_t decimals);
```
Change to `void setPacketData(float f, uint8_t decimals = 3);`

jgromes commented on src/Packet.cpp:
```diff
@@ -151,6 +151,44 @@ void Packet::copyInto(Packet& pack) {
+  strcpy(data, charArray);
+}
+
+void Packet::setPacketData(String wordy){
```
Change the argument to `String str`.

jgromes reviewed (changes requested):
Nice, I was thinking of implementing something similar myself, just never got around to actually doing it. Before I merge this, there's a few minor issues, I would be glad if you could address them.

I would like to also rewrite the example code, but I will do that myself after the merge (so that the writing style is consistent across all the examples).

Thanks!

renatocarusos committed:
Issues resolved.

1º. Unnecessary pointer "this->" removed; 2º. Corrected method overload; 3º. Unused variable removed; 4º. Default decimal value added in prototyping in .h; 5º. String "Wordy" changed to "str".

renatocarusos commented:
I was really worried about the style of the sample code. But I figured you'd stylize it according to your sketch.
